### ISSUE TITLE
In-Result Search Should Highlight Search Results In Graph View

### ISSUE THREAD

**Issue by lbrauns** [SpecterOps/BloodHound#279]
**Feature Description:**

When searching for an object on the graph and clicking on the search result, the object should be highlighted in the graph view.

**Current Behavior:**

Clicking on a search result only opens the detail box for the object.

**Desired Behavior:**

The highlights the search result in the graph or zooms to the object in the graph.

**Use Case:**

This will be especially useful in graphs like this (shortest path to domain admins):

![image](https://github.com/SpecterOps/BloodHound/assets/16474792/f4ac64d1-2e10-4af6-afa2-9cd147b10483)

It's also a nice example why https://github.com/SpecterOps/BloodHound/issues/89 would be really nice to have ;)

**Comment by StephenHinck** [comment]
Hi @lbrauns - the current functionality is that the object is highlighted, please see:

https://github.com/SpecterOps/BloodHound/assets/12970156/dfcce27e-8cc2-4675-b978-c190f05a07f1

We can modify the request to additionally zoom in on the object if you like?

**Comment by lbrauns** [comment]
Fair point... Unfortunately the highlighting is just not prominent enough in a bigger graph.

So the highlighting should be more prominent at least. Zooming to the node would be IMHO the better solution. Maybe a threshold of nodes in the graph could be used to distinguish, if the node should be highlighted or zoomed into.
